app/src/main/java/ch/epfl/sdp/blindly/database/UserRepositoryImpl.kt

```diff
@@ -181,24 +181,49 @@ class UserRepositoryImpl constructor(
 
 
     /**
-     * Removes another liked or matched user from current user.
+     * Removes a matched user from likes and matches and adds it to dislikes.
      *
-     * @param field field to remove a User (either from LIKES or MATCHES)
      * @param userId current user's ID
      * @param matchId matched user's ID
      */
-    override suspend fun removeMatchFromAUser(field: String, userId: String, matchId: String) {
-        var updatedList: ArrayList<String>? = arrayListOf()
+    override suspend fun removeMatchFromRemovingUser(
+        userId: String,
+        matchId: String
+    ) {
         val user = getUser(userId)
         if (user != null) {
-            when (field) {
-                LIKES ->
-                    updatedList = user.likes as ArrayList<String>?
-                MATCHES ->
-                    updatedList = user.matches as ArrayList<String>?
-            }
-            updatedList?.remove(matchId)
-            user.uid?.let { updateProfile(it, field, updatedList) }
+            var updatedLikesList = user.likes as ArrayList<String>?
+            var updatedDislikesList = user.dislikes as ArrayList<String>?
+            var updatedMatchesList = user.matches as ArrayList<String>?
+
+            updatedLikesList?.remove(matchId)
+            updatedDislikesList?.add(matchId)
+            updatedMatchesList?.remove(matchId)
+
+            user.uid?.let { updateProfile(it, LIKES, updatedLikesList) }
+            user.uid?.let { updateProfile(it, DISLIKES, updatedDislikesList) }
+            user.uid?.let { updateProfile(it, MATCHES, updatedMatchesList) }
+        }
+    }
+
+
+    /**
+     * Removes the current user from removed user's matches
+     * It's kept in likes of remote user so that they wouldn't reappear
```

Comment:
```suggestion
     * It's kept in likes of remote user so that they don't reappear
```

Reply:
:D Sure why not

Reply:
Don't know kinda souded strange xD (and I neede to find things to review x) )